app/authentication/server/startup/index.js

```diff
@@ -278,27 +276,22 @@ Accounts.insertUserDoc = _.wrap(Accounts.insertUserDoc, function (insertUserDoc,
 		}
 	}
 
-	if (noRoles || roles.length === 0) {
-		const hasAdmin = Users.findOne(
-			{
-				roles: 'admin',
-				type: 'user',
-			},
-			{
-				fields: {
-					_id: 1,
-				},
-			},
-		);
-
-		if (hasAdmin) {
-			roles.push('user');
-		} else {
+	/**
+	 * if settings shows setup wizard to be pending
+	 * and no admin's been found,
+	 * and existing role list doesn't include admin
+	 * create this user admin.
+	 * count this as the completion of setup wizard step 1.
+	 */
+	if (settings.get('Show_Setup_Wizard') === 'pending') {
```

Comment:
Using this as the "first" condition would only allow us promoting users to admin (when there are no other admins in the server) when the Setup Wizard is still pending, right? This is a slight change from what we were doing, but I'm afraid this could create a unwanted behavior by leaving the server with no admins 🤔 
Note that, previously, we could promote a new user to admin regardless of the setup wizard status.

Reply:
I though about that, but then iirc we don't allow deletion of the last admin via the app any way.

I also just tested it,

```
{
  "success": false,
  "error": "Leaving the app without admins is not allowed [error-action-not-allowed]",
  "errorType": "error-action-not-allowed",
  "details": {
    "method": "deleteUser",
    "action": "Remove_last_admin"
  }
}
```

The endpoint was called by a non-admin user with just user-delete permission.

This means the server can end up with no admin iff it's done directly by modifying the collection. In which case, rectifying would still be possible.

Reply:
> Doesn't seem like it's necessary, at least one admin's existence is enforced via the app/api.

Yup, agree. Still, I'm not sure we're not missing any edge case 🤔
I've just created #24308 based on your branch with some changes: I noticed we would miss the "user" role when creating users by the `userHandler.js` file, so I added this role as the default there. I've also added the `findOneByRolesAndType` method to improve legibility.
Please take a look there and feel free to request changes or approve it if you think it's ok 🙂

Reply:
> I noticed we would miss the "user" role when creating users by the `userHandler.js` file, so I added this role as the default there. I've also added the `findOneByRolesAndType` method to improve legibility.

That's not correct. Check issue that this PR intended to fix https://github.com/RocketChat/Rocket.Chat/issues/24148